### PR TITLE
refactor: fetch chemical phrases methods for merck

### PR DESCRIPTION
**Pull request by adambasha0**
- [x] rather 1-story 1-commit than sub-atomic commits

- [x] commit title is meaningful =>  git history search

- [x] commit description is helpful => helps the reviewer to understand the changes

- [x] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [x] added code is linted

- [x] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [ ] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [x] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

**Comment by adambasha0** [comment]
linked to ticket: https://helpdesk.nfdi4chem.de/#ticket/zoom/760

Problem description: fetching safety phrases for Merck problems does not work.
Reason of problem: internal Merck server changes.
solution: refactor fetch safety phrases methods for Merck

**Comment by adambasha0** [comment]
How to test:
[Chemotion dev (1).webm](https://github.com/user-attachments/assets/a4b41d96-a174-4c6f-8df3-56ee61ce6e01)

**Comment by haditariq** [comment]
1- On the very first time when Safety data sheet is deleted without saving it gives an error (attachment). 
merik crash on delete

**Comment by haditariq** [comment]
Button design can be changed because its not similiar to a standard button appearance. It has a scroll as well. 
buton desing

**Comment by adambasha0** [comment]
> On the very first time when Safety data sheet is deleted without saving it gives an error (attachment).

fixed

File: app/packs/src/components/ChemicalTab.js [hunk 1]
```diff
@@ -788,23 +789,24 @@ export default class ChemicalTab extends React.Component {
...
-          ) : (
-            <ListGroupItem key={`${key}-no-document`}>
-              <div>
-                <p className="safety-sheets-paragraph">
-                  {document}
-                </p>
-              </div>
-            </ListGroupItem>
-          )}
+          {document !== 'Could not find safety data sheet from Thermofisher'
```
Comment (haditariq) [inline app/packs/src/components/ChemicalTab.js:796]: May be wrapping both checks for "document" with a scope?

File: lib/chemotion/chemicals_service.rb [hunk 2]
```diff
@@ -88,7 +88,7 @@ def self.construct_h_statements(h_phrases, vendor = nil)
       h_statements = {}
       h_phrases_hash = JSON.parse(File.read('./public/json/hazardPhrases.json'))
 
-      h_array = vendor == 'merck' ? h_phrases[1].split(/\s[+-]\s/) : h_phrases
+      h_array = vendor == 'merck' ? h_phrases[4].split(',') : h_phrases
```
Comment (haditariq) [inline lib/chemotion/chemicals_service.rb:91]: Index 4 will always exists?

File: lib/chemotion/chemicals_service.rb [hunk 3]
```diff
@@ -98,7 +98,7 @@ def self.construct_h_statements(h_phrases, vendor = nil)
     def self.construct_p_statements(p_phrases, vendor = nil)
       p_statements = {}
       p_phrases_hash = JSON.parse(File.read('./public/json/precautionaryPhrases.json'))
-      p_array = vendor == 'merck' ? p_phrases[2].split(/\s[+-]\s/) : p_phrases
+      p_array = vendor == 'merck' ? p_phrases[5].split('-').map { |element| element.gsub(/\s+/, '') } : p_phrases
```
Comment (haditariq) [inline lib/chemotion/chemicals_service.rb:101]: Index 5 will always exists?

File: lib/chemotion/chemicals_service.rb [hunk 4]
```diff
@@ -135,11 +134,10 @@ def self.safety_section(product_link)
     def self.safety_phrases_merck(product_link)
       safety_section = safety_section(product_link)
       safety_array = safety_section.children.reject { |i| i.text.empty? }.map(&:text)
-      pictograms = safety_array[0].split(',')
-      verified_pictograms = pictograms.select { |pictogram| pictograms_hash.value?(pictogram) }
+      pictograms = safety_array[3].split(',')
```
Comment (haditariq) [inline lib/chemotion/chemicals_service.rb:137]: Index 3 will always exist?

**Review by haditariq — CHANGES_REQUESTED** [review]
Some questions and a few minor changes.
Thanks!

File: lib/chemotion/chemicals_service.rb (hunk 4 above)
Comment (adambasha0) [reply lib/chemotion/chemicals_service.rb:137]: yes, we are doing data mining using web scraping. This is how the server usually responds.

File: app/packs/src/components/ChemicalTab.js [hunk 5]
```diff
@@ -116,17 +116,18 @@ export default class ChemicalTab extends React.Component {
       safetySheets.splice(index, 1);
       this.setState({ safetySheets });
     }
-    if (chemical._chemical_data[0].safetySheetPath.length > 0) {
+    const path = chemical._chemical_data[0].safetySheetPath;
```
[the remark this answers is not in the document]
Comment (adambasha0) [reply app/packs/src/components/ChemicalTab.js:119]: already refactored

File: app/packs/src/components/ChemicalTab.js (hunk 1 above)
Comment (adambasha0) [reply app/packs/src/components/ChemicalTab.js]: done

**Pull request merged**